compliance/collection/inventory/convert.go

```diff
@@ -0,0 +1,110 @@
+package inventory
+
+import (
+	timestamp "github.com/gogo/protobuf/types"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/uuid"
+	scannerV1 "github.com/stackrox/scanner/generated/scanner/api/v1"
+)
+
+// ToNodeInventory converts a NodeInventory response to a native storage.NodeInventory version
+func ToNodeInventory(resp *scannerV1.GetNodeInventoryResponse) *storage.NodeInventory {
+	ni := &storage.NodeInventory{
+		NodeId:     uuid.Nil.String(),
+		NodeName:   resp.GetNodeName(),
+		ScanTime:   timestamp.TimestampNow(),
+		Components: toStorageComponents(resp.GetComponents()),
+		Notes:      convertNotes(resp.GetNotes()),
+	}
+	return ni
+}
+
+func toStorageComponents(c *scannerV1.Components) *storage.NodeInventory_Components {
+	if c == nil {
+		return nil
+	}
+
+	var namespace string
+	if c.GetNamespace() == "" {
+		namespace = "unknown"
+	} else {
+		namespace = c.GetNamespace()
+	}
+
+	sc := &storage.NodeInventory_Components{
+		Namespace:       namespace,
+		RhelComponents:  convertRHELComponents(c.GetRhelComponents()),
+		RhelContentSets: c.GetRhelContentSets(),
+	}
+	return sc
+
+}
+
+func convertRHELComponents(rc []*scannerV1.RHELComponent) []*storage.NodeInventory_Components_RHELComponent {
+	if rc == nil {
+		return nil
+	}
+	convertedComponents := make([]*storage.NodeInventory_Components_RHELComponent, 0)
+	for _, c := range rc {
+		sc := storage.NodeInventory_Components_RHELComponent{
+			Id:          c.GetId(),
+			Name:        c.GetName(),
+			Namespace:   c.GetNamespace(),
+			Version:     c.GetVersion(),
+			Arch:        c.GetArch(),
+			Module:      c.GetModule(),
+			AddedBy:     c.GetAddedBy(),
+			Executables: convertExecutables(c.GetExecutables()),
+		}
+		convertedComponents = append(convertedComponents, &sc)
+	}
+	return convertedComponents
+}
+
+func convertExecutables(e []*scannerV1.Executable) []*storage.NodeInventory_Components_RHELComponent_Executable {
+	if e == nil {
+		return nil
+	}
+	convertedExecutables := make([]*storage.NodeInventory_Components_RHELComponent_Executable, 0)
```

Comment:
Why not making setting the length of the slice and avoid letting go expand the slice at runtime? 
```suggestion
	convertedExecutables := make([]*storage.NodeInventory_Components_RHELComponent_Executable, len(e))
```